plinth/utils.py

```diff
@@ -52,6 +52,11 @@ def non_admin_view(func):
     return func
 
 
-def is_user_admin(user):
+def is_user_admin(request, cached=False):
     """Return whether user is an administrator."""
-    return user.groups.filter(name='admin').exists()
+    if cached and request.session['cache_user_is_admin'] is not None:
```

Comment:
@SunilMohanAdapa Thank you! I tested all the checks you mentioned, everything seems to work alright, except , each time I call the front page, as user, or admin, I see that a separate SQL query is made:
![image](https://cloud.githubusercontent.com/assets/20185076/24989766/99809cbe-202c-11e7-8427-26b4add15393.png)
Does this mean, caching is failing?